feat(ui): Show number of failing checks per group

quibusus opened the pull request:
Hi, Gatus is awesome.

I implemented a simple change to show a number of failing checks instead of just an icon.

Please let me know if interested; then I'll finish the tests, etc.



<!-- Thank you for contributing! -->

## Summary
<!-- If there's a relevant issue, you can just write the issue number below (e.g. #123) -->



## Checklist
<!-- Replace [ ] by [X] if you have completed the item -->
- [ ] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [ ] Updated documentation in `README.md`, if applicable.

TwiN reviewed (commented):
@quibusus Could you upload a screenshot? Just curious as to what it looks like.

quibusus commented:
Here you go:

<img width="1311" alt="dark-mode" src="https://github.com/TwiN/gatus/assets/965671/c0188b6b-b2f7-4a68-a016-0b9daf538d41">
<img width="1311" alt="light-mode" src="https://github.com/TwiN/gatus/assets/965671/b5019f02-bf9a-43a4-bcd2-d9368dd2ff58">


If the count is > 99, it displays the "!" as before. Showing 3 digits would break the design and is, IMHO, not happening often.

TwiN commented on web/app/src/components/EndpointGroup.vue:
```diff
@@ -7,11 +7,12 @@
           </span>
-          <span v-else class="float-right text-yellow-500 text-sm w-7 hover:scale-110" title="Partial Outage">
-            <ExclamationCircleIcon />
+          <span v-if="unhealthyCount==0" class="float-right text-green-600 w-7 hover:scale-110" title="Operational">
```
```suggestion
          <span v-if="unhealthyCount === 0" class="float-right text-green-600 w-7 hover:scale-110" title="Operational">
```

TwiN commented on web/app/src/components/EndpointGroup.vue:
```diff
@@ -7,11 +7,12 @@
-          <span v-if="healthy" class="float-right text-green-600 w-7 hover:scale-110" title="Operational">
-            <CheckCircleIcon />
+          <span v-if="unhealthyCount" class="float-right text-yellow-500 text-sm w-7 hover:scale-110" title="Partial Outage">
+            <span v-if="unhealthyCount<100" class="rounded-xl bg-yellow-500 text-gray-900 font-bold leading-6 float-right w-6 h-6 text-center hover:scale-110">{{unhealthyCount}}</span>
```
Why 100? Seems like an arbitrary number. Always displaying the number if it's non-0 seems like a better option imo

quibusus replied:
I chose 100 because it does not break design. More digits won't fit in the circle. I'll experiment with alternatives

quibusus replied:
updated, please review

TwiN replied:
Awesome, looks great!

quibusus commented:
<img width="1097" alt="Screenshot 2023-07-01 at 23 39 21" src="https://github.com/TwiN/gatus/assets/965671/f3acd5d2-6c92-41fc-9b4c-3df6a297d6d0">
<img width="1097" alt="Screenshot 2023-07-01 at 23 38 47" src="https://github.com/TwiN/gatus/assets/965671/e86182eb-33f2-4e83-8952-1437f7bd8b56">
<img width="1097" alt="Screenshot 2023-07-01 at 23 38 30" src="https://github.com/TwiN/gatus/assets/965671/a2cf845b-c6e3-428b-bb33-d8750cf7fd7d">
<img width="1097" alt="Screenshot 2023-07-01 at 23 38 19" src="https://github.com/TwiN/gatus/assets/965671/c2605bde-edf5-4d34-bc18-344b03564937">

updated design, please review

I removed the arbitrary number, simplified the code and changed to red - to be same as the failing "x"

TwiN commented:
Thank you for the contribution!
